libvast/builtins/operators/extend.cpp

```diff
@@ -152,6 +157,48 @@ class plugin final : public virtual pipeline_operator_plugin {
       return parsed_config.error();
     return std::make_unique<extend_operator>(std::move(*parsed_config));
   }
+
+  [[nodiscard]] std::pair<std::string_view,
+                          caf::expected<std::unique_ptr<pipeline_operator>>>
+  make_pipeline_operator(std::string_view pipeline) const override {
+    using parsers::optional_ws, parsers::required_ws, parsers::data,
+      parsers::end_of_pipeline_operator,
+      parsers::extractor_value_assignment_list;
+    const auto* f = pipeline.begin();
+    const auto* const l = pipeline.end();
+    const auto p = required_ws >> extractor_value_assignment_list >> optional_ws
+                   >> end_of_pipeline_operator;
+    std::vector<std::tuple<std::string, vast::data>> parsed_assignments;
+    if (!p(f, l, parsed_assignments)) {
+      return {
+        std::string_view{f, l},
+        caf::make_error(ec::syntax_error, fmt::format("failed to parse extend "
+                                                      "operator: '{}'",
+                                                      pipeline)),
+      };
+    }
+    record config_record;
+    record fields_record;
+    for (const auto& [key, data] : parsed_assignments) {
+      fields_record[key] = data;
+    }
+    config_record["fields"] = fields_record;
```

Comment:
std::move can be used here.
= std::move(fields_record)

I think it can also be used in the loop above but i think structured_bindings and std::move doesn't work that well so we can just ignore it as it isn't a hot path i guess.

Reply:
Implemented in https://github.com/tenzir/vast/pull/2877/commits/8545ed2393233c0027fb625c12d7fe3f30f9d6ca